### PR TITLE
Yubrew/patch-wl-price

### PR DESCRIPTION
**Pull request by yubrew**

### CODE REVIEW

File: contracts/minter/src/contract.rs [hunk 1]
```diff
@@ -332,8 +333,29 @@ pub fn execute_batch_mint(
         return Err(ContractError::MaxBatchMintLimitExceeded {});
     }
 
+    // calculate the mint fee
+    // if in whitelist, use whitelist price. else use mint price
+    let mint_price: Coin = if let Some(whitelist) = config.whitelist {
```
Comment (shanev) [inline contracts/minter/src/contract.rs:338]: Can break into a function?

**Review by shanev — APPROVED** [review]
Can probably break out common code into a function.

**Review by JakeHartnell — APPROVED** [review]

File: contracts/minter/src/contract.rs [hunk 2]
```diff
@@ -377,27 +399,48 @@ fn _execute_mint(
...
+            let unit_price: UnitPriceResponse = deps
+                .querier
+                .query_wasm_smart(whitelist, &WhitelistQueryMsg::UnitPrice {})?;
+            unit_price.unit_price
+        } else {
+            config.unit_price.clone()
+        }
+    } else {
+        config.unit_price.clone()
+    };
```
Comment (jhernandezb) [inline contracts/minter/src/contract.rs:420]: can we refactor into a method  seems like its the same code for batch and regular mint.

**Comment by shanev** [comment]
Sorry I merged this too quickly. Yes can refactor.

File: contracts/minter/src/contract.rs (hunk 2 above)
Comment (yubrew) [reply contracts/minter/src/contract.rs:420]: yes i was thinking the same!

**Pull request merged**